roles/aap_setup_install/tasks/main.yml

```
@@ -60,10 +60,10 @@
       until: __aap_setup_inst_result.status == 200
       retries: 90
       delay: 10
-      when: "'automationcontroller' in aap_setup_prep_inv_nodes"
+      when: "'automationcontroller' in aap_setup_prep_inv_nodes and aap_postinstall_validation|bool == true"
```

```suggestion
      when:
        - aap_setup_inst_validation | bool
        - 'automationcontroller' in aap_setup_prep_inv_nodes
```
A list of when's is combined with 'and' and more readable